app/models/patient.rb

```diff
@@ -79,15 +79,15 @@ class Patient < ApplicationRecord
   scope :search_by_address, ->(term) { joins(:address).merge(Address.search_by_street_or_village(term)) }
 
   scope :follow_ups_by_period, ->(period, at_region: nil, current: true, last: nil) {
-    FollowUpsQuery.with(Encounter, period, at_region: at_region, current: current, time_column: "encountered_on", last: last)
+    FollowUpsQuery.new.with(Encounter, period, at_region: at_region, time_column: "encountered_on", current: current, last: last, time_zone: false)
```

Comment:
The `time_zone: false` is the crux of the bugfix. The rest is cleanup and refactoring.